rust/chains/hyperlane-cosmos/src/providers/rpc.rs

```diff
@@ -1,29 +1,27 @@
-use std::ops::RangeInclusive;
-
 use async_trait::async_trait;
 use cosmrs::rpc::client::Client;
-use cosmrs::rpc::endpoint::{tx, tx_search::Response as TxSearchResponse};
-use cosmrs::rpc::query::Query;
-use cosmrs::rpc::Order;
-use cosmrs::tendermint::abci::EventAttribute;
 use hyperlane_core::{ChainCommunicationError, ChainResult, ContractLocator, LogMeta, H256, U256};
-use tracing::{instrument, trace};
+use sha256::digest;
+use tendermint::hash::Algorithm;
+use tendermint::Hash;
+use tendermint_rpc::endpoint::block::Response as BlockResponse;
+use tendermint_rpc::endpoint::block_results::Response as BlockResultsResponse;
+use tracing::{debug, instrument, trace};
 
 use crate::address::CosmosAddress;
+use crate::payloads::general::{EventAttribute, Events};
 use crate::{ConnectionConf, CosmosProvider, HyperlaneCosmosError};
 
-const PAGINATION_LIMIT: u8 = 100;
-
 #[async_trait]
 /// Trait for wasm indexer. Use rpc provider
 pub trait WasmIndexer: Send + Sync {
     /// Get the finalized block height.
     async fn get_finalized_block_number(&self) -> ChainResult<u32>;
 
     /// Get logs for the given range using the given parser.
-    async fn get_range_event_logs<T>(
+    async fn get_event_log<T>(
```

Comment:
nit: iirc this was the name from mitosis before but it feels a bit off, maybe `get_logs_in_block` or something? also need to update the comment above